src/python_bindings/fastvdf.cpp

```diff
@@ -61,6 +61,24 @@ PYBIND11_MODULE(chiavdf, m) {
         return is_valid;
     });
 
+    // Checks an N wesolowski proof.
+    m.def("create_discriminant_and_verify_n_wesolowski", [] (const py::bytes& challenge_hash,
+                                   const int discriminant_size_bits,
+                                   const string& x_s,
+                                   const string& proof_blob,
+                                   const uint64_t num_iterations,
+                                   const uint64_t recursion) {
+        std::string challenge_hash_str(challenge_hash);
+        std::vector<uint8_t> challenge_hash_bits = std::vector<uint8_t>(challenge_hash_str.begin(), challenge_hash_str.end());
+        std::string x_s_copy(x_s);
+        std::string proof_blob_copy(proof_blob);
+        bool is_valid = false;
+        {
```

Comment:
Introducing this block here will only make you acquire the GIL slightly earlier. But you don't actually need the GIL to be acquired earlier. So, I think it's only has drawbacks. more lines of code, giving the reader the impression that it's important and acquiring the GIL earlier than necessary.

returning `is_valid` is not interacting with the python interpreter.

Reply:
it is returned back to python land though, so i wanted to make sure the GIL was on. just trying to assume nothing.